lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -94,8 +94,13 @@ impl<TGraphLinks: GraphLinks> HNSWIndex<TGraphLinks> {
         } else {
             let vector_storage = vector_storage.borrow();
             let available_vectors = vector_storage.available_vector_count();
-            let full_scan_threshold = hnsw_config.full_scan_threshold.saturating_mul(BYTES_IN_KB)
-                / (vector_storage.vector_dim() * VECTOR_ELEMENT_SIZE);
+            let full_scan_threshold = if available_vectors > 0 {
+                hnsw_config.full_scan_threshold.saturating_mul(BYTES_IN_KB)
+                    / (vector_storage.available_size_in_bytes()
```

Comment:
Are we sure this one is never 0?

Reply:
There is check for 0. If 0, `full_scan_threshold` will be 1 as a placeholder value. It's a case of indexed empty segment. It may happen when user has multiple names but didn't set any vector value for some name. Any placeholder value is valid id guess